models/transaction.go

```diff
@@ -11,30 +14,28 @@ func NewGormTransactionSupport(db *gorm.DB) *GormTransactionSupport {
 
 // GormTransactionSupport implements TransactionSupport for gorm
 type GormTransactionSupport struct {
-	tx *gorm.DB
 	db *gorm.DB
 }
 
-func (g *GormTransactionSupport) TX() *gorm.DB {
-	return g.tx
-}
-
 // Begin implements TransactionSupport
-func (g *GormTransactionSupport) Begin() error {
-	g.tx = g.db.Begin()
-	return g.db.Error
+func (g *GormTransactionSupport) Begin() (transaction.Transaction, error) {
+	tx := g.db.Begin()
+	return tx, g.db.Error
```

Comment:
You are correct. How great it would be if stuff like that was actually documented.